src/main/AppWithRouter.js

```diff
@@ -35,7 +35,7 @@ export default class App extends PureComponent {
             <Menu
               theme="dark"
               mode="horizontal"
-              defaultSelectedKeys={[selectKey]}
+              selectedKeys={[selectKey]}
```

Comment:
My understanding of antd Menu is that `defaultSelectedKeys` is only for initial state. `selectedKeys` is for current state. See https://ant.design/components/menu/#Menu

I have confirmed that this fixes the issue where the navigation is out of sync with the content.